tests/test_player.py

```diff
@@ -0,0 +1,303 @@
+            assert isinstance(buff, Player)
+            assert buff.batting_stars <= player.batting_stars
+            assert buff.patheticism >= player.patheticism
+            if getattr(player, "cinnamon", None):
+                assert buff.cinnamon > player.cinnamon
+
+            reroll = player.simulated_copy(reroll={"overall_rating": True, "patheticism": True, "cinnamon": True})
+            assert isinstance(reroll, Player)
+
+            override = player.simulated_copy(overrides={"cinnamon": 6.9})
+            assert isinstance(override, Player)
+            assert override.cinnamon == 6.9
+
+    def test_stars_bounded(self):
+        """
+        Test that star equations produce correct results
+        """
+        with open(f"{TEST_DATA_DIR}/player/stars.json", "r") as fp:
+            players = [Player(data) for data in json.load(fp)]
+            for player in players:
+                assert player.batting_stars == pytest.approx(player.reference_batting_stars)
+                assert player.pitching_stars == pytest.approx(player.reference_pitching_stars)
+                assert player.baserunning_stars == pytest.approx(player.reference_baserunning_stars)
+                assert player.defense_stars == pytest.approx(player.reference_defense_stars)
+
+    def test_soulscream_bounded(self):
+        """
+        Test that soulscream equation produces correct results
+        """
+        with open(f"{TEST_DATA_DIR}/player/screams.json", "r") as fp:
+            players = [Player(data) for data in json.load(fp)]
+            for player in players:
+                assert player.soulscream == pytest.approx(player.reference_scream)
+
+    def test_vibes_bounded(self):
+        """
+        Test that vibe equation produces correct results
+        """
+        with open(f"{TEST_DATA_DIR}/player/vibes.json", "r") as fp:
+            players = [Player(data) for data in json.load(fp)]
+            for player in players:
+                for day in range(1, 100):
+                    assert player.get_vibe(day) == pytest.approx(player.vibes[day-1])
+
+    @pytest.mark.vcr
+    def test_blood_coffee_bounded(self):
+        """
+        Test that blood & coffee produces the correct results
+        """
+        with open(f"{TEST_DATA_DIR}/player/blood_coffee.json", 'r') as fp:
+            players = [Player(data) for data in json.load(fp)]
+            for player in players:
+                assert player.blood == player.reference_blood
+                assert player.coffee == player.reference_coffee
+
+    @pytest.mark.vcr
+    def test_long_soulscream(self):
+        """
+        Test very long soulscreams
+        """
+        test_player = Player.find_by_name("Chorby Soul")
+        test_value = test_player.soulscream
+        assert isinstance(test_value, str)
+        assert test_value[-9:] == "undefined"
+
+    def test_player_make_random(self):
```

Comment:
separate these into two different test cases. consider moving Blintz from json to a fixture in conftest